src/modules/navigator/MissionFeasibility/FeasibilityChecker.hpp

```diff
@@ -0,0 +1,139 @@
+ *
+ *   Copyright (c) 2022 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#pragma once
+
+#include "../navigation.h"
+#include <mathlib/mathlib.h>
+#include <uORB/topics/home_position.h>
+#include <uORB/topics/vehicle_status.h>
+#include <uORB/topics/vehicle_land_detected.h>
+#include <uORB/Subscription.hpp>
+#include <px4_platform_common/module_params.h>
+
+class FeasibilityChecker : public ModuleParams
+{
+public:
+
+	FeasibilityChecker();
+
+	void processNextItem(mission_item_s &mission_item, const int current_index, const int total_count);
+
+	void setMavlinkLogPub(orb_advert_t *mavlink_log_pub)
+	{
+		_mavlink_log_pub = mavlink_log_pub;
+	}
+
+	bool someCheckFailed()
+	{
+		return _takeoff_failed ||
+		       _distance_first_waypoint_failed ||
+		       _distance_between_waypoints_failed ||
+		       _fixed_wing_landing_failed ||
+		       _below_home_alt_failed ||
+		       _mission_validity_failed ||
+		       _takeoff_land_available_failed;
+	}
+
+	void reset();
+
+private:
+	orb_advert_t *_mavlink_log_pub{nullptr};
+
+	uORB::Subscription _home_pos_sub{ORB_ID(home_position)};
+	uORB::Subscription _status_sub{ORB_ID(vehicle_status)};
+	uORB::Subscription _land_detector_sub{ORB_ID(vehicle_land_detected)};
+
+	DEFINE_PARAMETERS(
+		(ParamFloat<px4::params::FW_LND_ANG>)   _param_fw_lnd_ang,
```

Comment:
You'll need to be careful with param enums here, this needs to build in boards that might not have the FW modules at all.

Reply:
addressed